draft-ietf-quic-transport.md

```diff
@@ -6357,6 +6357,217 @@ this behavior.  An endpoint can then immediately close the connection with a
+ciphertext. An attack is successful if an attacker can cause a peer to send a
+UDP datagram to a host that will perform some action based on content in the
+datagram that the attacker is able to control or influence.
+
+This section discusses ways in which QUIC might be used for request forgery
+attacks and describes countermeasures that can be implemented by QUIC
+endpoints. These actions are described on the assumption that potential targets
+for request forgery attacks take no action to protect against these attacks.
+While it would be ideal if target services implement better protections, such
+as strong authentication that does not rely on implicit signals, the goal of
+this section is to describe what a QUIC implementation or deployment can do.
+
+
+### Control Options for Endpoints
+
+QUIC offers some opportunities for endpoints to influence or control where
+their peers send UDP datagrams.
+
+Control over the destination of UDP datagrams presents three options for
+request forgery attack:
+
+* initial connection establishment ({{handshake}}), where a server is able to
+  choose where a client sends datagrams, for example by populating DNS records;
+
+* preferred addresses ({{preferred-address}}), where a server is able to choose
+  where a client sends datagrams; and
+
+* spoofed connection migrations ({{address-spoofing}}), where a client is able
+  to use source address spoofing to select where a server sends subsequent
+  datagrams.
+
+These three options all involve cases where an endpoint sends datagrams to a
+host that might not understand QUIC. That is, these packets are sent prior to
+address validation; see {{address-validation}}.
+
+Outside of the encrypted portion of packets, QUIC offers an endpoint several
+options for controlling the content of UDP datagrams that a peer sends. The
+Destination Connection ID field offers direct control over early bytes of every
+packet sent by a peer; see {{connection-id}}. The Token field in Initial
+packets offers a server control over other bytes of Initial packets; see
+{{packet-initial}}.
+
+There are no measures in the protocol to prevent indirect control over the
+encrypted portions of packets. It is necessary to assume that endpoints are
+able to control the contents of frames that a peer sends, especially those
+frames that convey application data, such as STREAM frames. Though this depends
+to some degree on details of the application protocol, some control is possible
+in many protocol usage contexts. As the attacker has access to packet protection
+keys, they are likely to be capable of predicting how a peer will encrypt future
+packets. Successful control over datagram content then only requires that the
+attacker be able to predict the packet number and placement of frames in packets
+with some amount of reliability.
+
+This section assumes that limiting control over datagram content is not
+feasible. The focus of the mitigations in subsequent sections is on limiting
```

Comment:
I am assuming you mean to say that limiting this for the purposes of mitigating is not feasible. Perhaps phrase it as "It is reasonable to assume that mitigations that require limitations on datagram content are not feasible. Therefore, the focus of the mitigations ..."

Reply:
It would also be reasonable to implement some countermeasures that prevented control over datagram content.  The ciphertext fix does costs a lot though (it's a major change to the protocol that likely has a significant performance impact or requires the use of new ciphers) and it is not sufficient either as there are other more serious problems, like the SPA+connection ID one that I think we've agreed is too valuable a feature to dispense with (and options without this exposure are likely intractable for some server deployments).

I could write all of that down, but I thought it better to just state the assumption.